### PR TITLE
Refactor `before` filters for `index` action of the webui request controller

### PR DESCRIPTION
**Pull request by eduardoj**

### CODE REVIEW

File: src/api/app/controllers/webui/request_controller.rb [hunk 1]
```diff
@@ -30,13 +28,20 @@ class Webui::RequestController < Webui::WebuiController
...
   before_action :redirect_to_tasks, only: [:index], unless: -> { Flipper.enabled?(:request_index, User.session) }
-  before_action :set_requests, :set_filter_involvement, :set_filter_state, :set_filter_action_type, :set_filter_creators,
-                :filter_requests, :set_selected_filter, only: [:index], if: lambda {
-                                                                              Flipper.enabled?(:request_index, User.session)
-                                                                            }
+
   after_action :verify_authorized, only: [:create]
 
   def index
+    if Flipper.enabled?(:request_index, User.session)
```
Comment (hennevogel) [inline src/api/app/controllers/webui/request_controller.rb:35]: Is there even an index action outside of beta? :)

Comment (eduardoj) [reply src/api/app/controllers/webui/request_controller.rb]: Nop, there isn't any. Removed. :+1:

File: src/api/app/controllers/webui/request_controller.rb [hunk 2]
```diff
@@ -29,14 +27,18 @@ class Webui::RequestController < Webui::WebuiController
...
   after_action :verify_authorized, only: [:create]
 
   def index
+    set_filter_involvement
+    set_filter_state
+    set_filter_action_type
+    set_filter_creators
+
+    filter_requests
+    set_selected_filter
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/controllers/webui/request_controller.rb:40]: Not much in the end, from the functional perspective.

Usually, we use `before_action` filters to call methods shared between actions of the same controller.

This is the opposite case. Moving method calls from `before_action` filters to inside a particular action makes it more clear this method calls are used by only one action. We already have lots of `before_action` filters.

File: src/api/app/controllers/concerns/webui/requests_filter.rb [hunk 3]
```diff
@@ -1,8 +1,28 @@
...
+  end
+
+  def set_filter_action_type
+    @filter_action_type = params[:action_type].presence || []
+    @filter_action_type = @filter_action_type.intersection(BsRequestAction::TYPES)
+  end
+
+  def set_filter_creators
+    @filter_creators = params[:creators].presence || []
+  end
```
Comment (danidoni) [inline src/api/app/controllers/concerns/webui/requests_filter.rb:24]: Looks like those methods are not reused anywhere else, I would just inline them in `index`

**Review by danidoni — APPROVED** [review]

File: src/api/app/controllers/concerns/webui/requests_filter.rb (hunk 3 above)
Comment (eduardoj) [reply src/api/app/controllers/concerns/webui/requests_filter.rb:24]: Just take a look at the description of the last commit:

_This way, they can be reused by other pages with requests filters._

Comment (danidoni) [reply src/api/app/controllers/concerns/webui/requests_filter.rb:24]: Oh, wait, this is going to be reused later for package and project right? Nvm then

**Pull request merged**